src/commands/runDraftTool/helper/runDraftHelper.ts

```diff
@@ -59,19 +62,22 @@ export async function downloadDraftBinary() {
    if (failed(downloadResult)) {
       return {
          succeeded: false,
-         error: [`Failed to download draft binary: ${downloadResult.error[0]}`]
+         error: `Failed to download draft binary: ${downloadResult.error} from ${draftDownloadUrl}`
       };
    }
    //If linux check -- make chmod 0755
    fs.chmodSync(destinationFile, '0755');
-   return succeeded(downloadResult);
+   return {
+      succeeded: true,
+      result: null
+   };
 }
 
 function getBinaryFileName() {
    let architecture = os.arch();
    let operatingSystem = os.platform().toLocaleLowerCase();
 
-   if (architecture === 'x64') {
+   if (architecture === 'x64' || architecture === 'ia32') {
```

Comment:
💡 Please note\Question: I don't this binary of x64 arch will be able to run on x32 the draft tool is generated for binaries for x32 arch? reason I ask is that this `if` is suggesting that hence asking for clarity. Thank you.

Reply:
i had the same thought, but this was required to get the tests to run on the github smoke test runner

Reply:
There are 2 things here: (please help me to know a little more around this) thanks

1) I see it must be some required thing hut can we first check what is it? (And not make changes which might not be logically correct or might be out of place reason I say this is because form logic above we are assuming x64 and x32 both will work form same binary is it true )?

2) does current state of draft-tool publish x32 binaries? 

Thanks heaps,

Reply:
https://github.com/Azure/aks-devx-tools/actions/runs/3715528818/jobs/6300805074#step:6:130

Reply:
💡☕️ 💼 Cool, thanks so this is just a runner running on all Arch - for correct contextual preference here is what we should think before adding wrong context: 

* There is something called `runner.arch` which should give `x64` et. al. and we should skip this for `x32` here is the link which will help: https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context 

* Doing this we will not fall into the trap of brut forcing something to work if Draft tool don't support x32 binary. I will stay around for your ping for quick discussion as well and we can chat this there. 
 
Thanks heaps.